### PR TITLE
Alerting: MuteTiming service return errutil + GetTiming by name

### PR DESCRIPTION
**Pull request by yuri-tceretian**
**What is this feature?**
This PR refactors MuteTiming service to return errutil.Error that contains statuses.

- Adds a new method to get mute timing by name and removes this logic from the request handler.
- Replaces errors with errutil.Error that is resolved to response status. It simplifies the API request handler logic.

- Updates method UpdateMuteTiming to return an error if mute timing is not found by name. Currently, it returns nil, and nil is translated to a NotFound response by API. This does not change the file-provisioning logic because it determines operation based on the current config.

Note: All errors mention `time interval` instead `mute timing` because the latter is deprecated and will be removed in the future. Also, the new "regular" API will use time interval semantic and will reuse the errors.

**Why do we need this feature?**
This will let us reuse this service in a new non-provisioning API and avoid duplication of logic and necessity of separate unit-tests. 

**Who is this feature for?**
- Developers
- Introduction of errutil errors let's us to return more verbose errors and error IDs, which less users (and UI in the future) handle exceptional cases better. 

**Special notes for your reviewer:**

- Based on https://github.com/grafana/grafana/pull/79817 and https://github.com/grafana/grafana/pull/79814
- To see the changes more clearly, you can review PR by commit.

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: pkg/services/ngalert/api/api_provisioning.go [hunk 1]
```diff
@@ -243,22 +244,17 @@ func (srv *ProvisioningSrv) RouteDeleteTemplate(c *contextmodel.ReqContext, name
...
 	}
-	return response.Empty(http.StatusNotFound)
+	return response.JSON(http.StatusOK, timing)
 }
 
 func (srv *ProvisioningSrv) RouteGetMuteTimingExport(c *contextmodel.ReqContext, name string) response.Response {
 	timings, err := srv.muteTimings.GetMuteTimings(c.Req.Context(), c.SignedInUser.GetOrgID())
 	if err != nil {
-		return ErrResp(http.StatusInternalServerError, err, "")
+		return response.ErrOrFallback(http.StatusInternalServerError, "failed to get mute timings", err)
```
Comment (rwwiv) [inline pkg/services/ngalert/api/api_provisioning.go:257]: `mute timings -> time intervals`

File: pkg/services/ngalert/api/api_provisioning.go [hunk 2]
```diff
@@ -272,15 +268,15 @@ func (srv *ProvisioningSrv) RouteGetMuteTimingExport(c *contextmodel.ReqContext,
 func (srv *ProvisioningSrv) RouteGetMuteTimings(c *contextmodel.ReqContext) response.Response {
 	timings, err := srv.muteTimings.GetMuteTimings(c.Req.Context(), c.SignedInUser.GetOrgID())
 	if err != nil {
-		return ErrResp(http.StatusInternalServerError, err, "")
+		return response.ErrOrFallback(http.StatusInternalServerError, "failed to get mute timings", err)
```
Comment (rwwiv) [inline pkg/services/ngalert/api/api_provisioning.go:271]: `mute timings -> time interval`

File: pkg/services/ngalert/api/api_provisioning.go [hunk 3]
```diff
@@ -272,15 +268,15 @@ func (srv *ProvisioningSrv) RouteGetMuteTimingExport(c *contextmodel.ReqContext,
...
 	}
 	return response.JSON(http.StatusOK, timings)
 }
 
 func (srv *ProvisioningSrv) RouteGetMuteTimingsExport(c *contextmodel.ReqContext) response.Response {
 	timings, err := srv.muteTimings.GetMuteTimings(c.Req.Context(), c.SignedInUser.GetOrgID())
 	if err != nil {
-		return ErrResp(http.StatusInternalServerError, err, "")
+		return response.ErrOrFallback(http.StatusInternalServerError, "failed to get mute timings", err)
```
Comment (rwwiv) [inline pkg/services/ngalert/api/api_provisioning.go:279]: `mute timings -> time intervals`

File: pkg/services/ngalert/api/api_provisioning.go [hunk 4]
```diff
@@ -290,10 +286,7 @@ func (srv *ProvisioningSrv) RoutePostMuteTiming(c *contextmodel.ReqContext, mt d
 	mt.Provenance = determineProvenance(c)
 	created, err := srv.muteTimings.CreateMuteTiming(c.Req.Context(), mt, c.SignedInUser.GetOrgID())
 	if err != nil {
-		if errors.Is(err, provisioning.ErrValidation) {
-			return ErrResp(http.StatusBadRequest, err, "")
-		}
-		return ErrResp(http.StatusInternalServerError, err, "")
+		return response.ErrOrFallback(http.StatusInternalServerError, "failed to create mute timing", err)
```
Comment (rwwiv) [inline pkg/services/ngalert/api/api_provisioning.go:289]: `mute timing -> time interval`

File: pkg/services/ngalert/api/api_provisioning.go [hunk 5]
```diff
@@ -243,22 +244,17 @@ func (srv *ProvisioningSrv) RouteDeleteTemplate(c *contextmodel.ReqContext, name
...
-	timings, err := srv.muteTimings.GetMuteTimings(c.Req.Context(), c.SignedInUser.GetOrgID())
+	timing, err := srv.muteTimings.GetMuteTiming(c.Req.Context(), name, c.SignedInUser.GetOrgID())
 	if err != nil {
-		return ErrResp(http.StatusInternalServerError, err, "")
-	}
-	for _, timing := range timings {
-		if name == timing.Name {
-			return response.JSON(http.StatusOK, timing)
-		}
+		return response.ErrOrFallback(http.StatusInternalServerError, "failed to get mute timing by name", err)
```
Comment (rwwiv) [inline pkg/services/ngalert/api/api_provisioning.go:249]: `mute timing -> time interval`

File: pkg/services/ngalert/api/api_provisioning.go [hunk 6]
```diff
@@ -303,21 +296,15 @@ func (srv *ProvisioningSrv) RoutePutMuteTiming(c *contextmodel.ReqContext, mt de
...
+		return response.ErrOrFallback(http.StatusInternalServerError, "failed to update mute timing", err)
 	}
 	return response.JSON(http.StatusAccepted, updated)
 }
 
 func (srv *ProvisioningSrv) RouteDeleteMuteTiming(c *contextmodel.ReqContext, name string) response.Response {
 	err := srv.muteTimings.DeleteMuteTiming(c.Req.Context(), name, c.SignedInUser.GetOrgID())
 	if err != nil {
-		return ErrResp(http.StatusInternalServerError, err, "")
+		return response.ErrOrFallback(http.StatusInternalServerError, "failed to delete mute timing", err)
```
Comment (rwwiv) [inline pkg/services/ngalert/api/api_provisioning.go:307]: `mute timing -> time interval`

File: pkg/services/ngalert/api/tooling/api.json [hunk 7]
```diff
@@ -5461,6 +5461,10 @@
     "responses": {
      "204": {
       "description": " The mute timing was deleted successfully."
+     },
```
Comment (rwwiv) [inline pkg/services/ngalert/api/tooling/api.json:5464]: `mute timing -> time interval`

File: pkg/services/ngalert/api/tooling/definitions/provisioning_mute_timings.go [hunk 8]
```diff
@@ -63,6 +63,7 @@ import (
 //
 //     Responses:
 //       204: description: The mute timing was deleted successfully.
+//       409: GenericError description: The mute timing is in use
```
Comment (rwwiv) [inline pkg/services/ngalert/api/tooling/definitions/provisioning_mute_timings.go:66]: `mute timing -> time interval`

**Review by rwwiv — APPROVED** [review]
Noted some string references to mute timing(s), rest of the PR looks good!

**Comment by yuri-tceretian** [comment]
Thanks! The reason I kept "mute-timings" in those places is because that's mute timing provisioning API whereas service will be used in a new time interval API, and eventually, mute timings will be deprecated (hopefully in G11)

**Pull request merged**